### PR TITLE
Require a trait in the bounds of existential types

### LINKED ISSUE #53090: `existential type` can accept generic bounds that do not specify a trait
```rust
#![feature(existential_type)]

existential type Foo: 'static; // ok

fn foo() -> Foo {
    "foo"
}

fn bar() -> impl 'static { // error: at least one trait must be specified
    "foo"
}

fn main() {}
```

Although the [RFC](https://github.com/rust-lang/rfcs/pull/2071) wasn't explicit about this, it seems that this is an oversight, as `existential type` should function like return-position `impl Trait`.

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #53090 

r? @oli-obk

### CODE REVIEW

File: src/test/ui/existential_types/existential-types-with-no-traits.stderr [hunk 1]
```diff
@@ -0,0 +1,14 @@
+error: at least one trait must be specified
+  --> $DIR/existential-types-with-no-traits.rs:3:1
+   |
+LL | existential type Foo: 'static;
```
Comment (oli-obk) [inline src/test/ui/existential_types/existential-types-with-no-traits.stderr]: can you limit the span to just the bounds? Maybe iterate over the bounds and collect all the spans into one `Vec` and use https://doc.rust-lang.org/nightly/nightly-rustc/syntax/source_map/struct.MultiSpan.html#method.from_spans to create a multispan. Then use that multispan to report the error.

Comment (JohnTitor) [reply src/test/ui/existential_types/existential-types-with-no-traits.stderr]: Okay, I used multispan instead.

**Comment by oli-obk** [comment]
@bors r+

Thanks!

**Comment by JohnTitor** [comment]
Oops, I didn't check the `run-pass` tests. Will fix them.

**Comment by JohnTitor** [comment]
@oli-obk Could we remove these lines, or should fix? (These are no longer `run-pass`.)

https://github.com/rust-lang/rust/blob/f843ad60ef5a78d8b4da85c5007d3ecac229f1a8/src/test/run-pass/existential_type.rs#L70-L83

**Comment by oli-obk** [comment]
@JohnTitor keep them as `run-pass`, but change the bounds to `Sized + 'a` or `Sized + 'static` respectively, which will amount to the same thing that we had before and should the tests compile.

**Comment by JohnTitor** [comment]
Thank you @oli-obk! I fixed them.

**Comment by JohnTitor** [comment]
Travis with green

**Comment by oli-obk** [comment]
@bors r+

Thanks!

**Pull request merged**